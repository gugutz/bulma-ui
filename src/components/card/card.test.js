import React from 'react'
import renderer from 'react-test-renderer'
import Image from '../image/image'
import Content from '../content/content'
import Media from '../media/media'
import Card from './card'

describe('<Card /> component', () => {
  it('should render correctly card with image', () => {
    const tree = renderer
      .create(
        <Card>
          <Card.Image>
            <Image
              size="4by3"
              alt="Placeholder image"
              src="https://bulma.io/images/placeholders/1280x960.png"
            />
          </Card.Image>
          <Card.Content>
            <Media>
              <Media.Left>
                <Image
                  size="48x48"
                  alt="Placeholder image"
                  src="https://bulma.io/images/placeholders/96x96.png"
                />
              </Media.Left>
              <Media.Content>
                <p className="title is-4">John Smith</p>
                <p className="subtitle is-6">@johnsmith</p>
              </Media.Content>
            </Media>
            <Content>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                Phasellus nec iaculis mauris. @bulmaio. #css #responsive
              </p>
              <time dateTime="2016-1-1">11:09 PM - 1 Jan 2016</time>
            </Content>
          </Card.Content>
        </Card>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly card with title', () => {
    const tree = renderer
      .create(
        <Card>
          <Card.Header>
            <Card.Header.Title>Component</Card.Header.Title>
            <Card.Header.Icon>x</Card.Header.Icon>
          </Card.Header>
          <Card.Content>
            <Content>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                Phasellus nec iaculis mauris. @bulmaio. #css #responsive
              </p>
              <time dateTime="2016-1-1">11:09 PM - 1 Jan 2016</time>
            </Content>
          </Card.Content>
          <Card.Footer>
            <Card.Footer.Item>Save</Card.Footer.Item>
            <Card.Footer.Item>Edit</Card.Footer.Item>
            <Card.Footer.Item>Delete</Card.Footer.Item>
          </Card.Footer>
        </Card>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render correctly card with title centered', () => {
    const tree = renderer
      .create(
        <Card>
          <Card.Header>
            <Card.Header.Title isCentered>Component</Card.Header.Title>
          </Card.Header>
          <Card.Content>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus
              nec iaculis mauris. @bulmaio. #css #responsive
            </p>
          </Card.Content>
        </Card>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
