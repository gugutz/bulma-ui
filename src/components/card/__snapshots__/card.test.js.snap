// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> component should render correctly card with image 1`] = `
<div
  className="card"
>
  <div
    className="card-image"
  >
    <figure
      className="image is-4by3"
    >
      <img
        alt="Placeholder image"
        src="https://bulma.io/images/placeholders/1280x960.png"
      />
    </figure>
  </div>
  <div
    className="card-content"
  >
    <div
      className="media"
    >
      <div
        className="media-left"
      >
        <figure
          className="image is-48x48"
        >
          <img
            alt="Placeholder image"
            src="https://bulma.io/images/placeholders/96x96.png"
          />
        </figure>
      </div>
      <div
        className="media-content"
      >
        <p
          className="title is-4"
        >
          John Smith
        </p>
        <p
          className="subtitle is-6"
        >
          @johnsmith
        </p>
      </div>
    </div>
    <div
      className="content"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec iaculis mauris. @bulmaio. #css #responsive
      </p>
      <time
        dateTime="2016-1-1"
      >
        11:09 PM - 1 Jan 2016
      </time>
    </div>
  </div>
</div>
`;

exports[`<Card /> component should render correctly card with title 1`] = `
<div
  className="card"
>
  <header
    className="card-header"
  >
    <p
      className="card-header-title"
    >
      Component
    </p>
    <a
      className="card-header-icon"
      type="button"
    >
      x
    </a>
  </header>
  <div
    className="card-content"
  >
    <div
      className="content"
    >
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec iaculis mauris. @bulmaio. #css #responsive
      </p>
      <time
        dateTime="2016-1-1"
      >
        11:09 PM - 1 Jan 2016
      </time>
    </div>
  </div>
  <footer
    className="card-footer"
  >
    <p
      className="card-footer-item"
      type="button"
    >
      Save
    </p>
    <p
      className="card-footer-item"
      type="button"
    >
      Edit
    </p>
    <p
      className="card-footer-item"
      type="button"
    >
      Delete
    </p>
  </footer>
</div>
`;

exports[`<Card /> component should render correctly card with title centered 1`] = `
<div
  className="card"
>
  <header
    className="card-header"
  >
    <p
      className="card-header-title is-centered"
    >
      Component
    </p>
  </header>
  <div
    className="card-content"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec iaculis mauris. @bulmaio. #css #responsive
    </p>
  </div>
</div>
`;
